### ISSUE TITLE
trying to add local repo makes a new repo under my windows profile

### ISSUE THREAD

**Issue by SarahGG** [desktop/desktop#8511]
### Describe the bug

I do to file > add local repository > and selected an existing folder at local path `C:\Users\sguarino\Documents\GitHub\WebDeveloperBootcampByColtSteele`

It then proceeds to add the git files and start tracking everything under `C:\Users\sguarino\` instead, effectively creating a repo for my entire user account.... all 32,752 changes worth.

### Version & OS

I am using GitHub version 2.2.1 on Windows 10 Enterprise version 1903.

### Steps to reproduce the behavior

1. Add folder named `WebDeveloperBootcampByColtSteele` under `your_chosen_class_path\GitHub\`.
2. In the GitHub Desktop App, select File > Add Local Repository
3. Select your `WebDeveloper...` folder
4. It then adds the .gitignore and everything under `C:\Users\your_username` instead

### Expected behavior

I expected it to add the folder as an existing repo

### Actual behavior

it turned my entire user folder into a repo instead

### Screenshots

Please see imgur folder of these steps screenshots: https://imgur.com/gallery/WHA72EM

### Logs

Attach your logs by opening the `Help` menu and selecting `Show Logs...`, if applicable.

```
***************your system isn't allowing me to attach files. Here it is copy and pasting
2019-10-22T00:24:24.321Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T00:39:52.382Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T01:12:09.809Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T01:44:27.272Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T11:31:54.766Z - info: [ui] [BranchPruner] Last prune took place in 12 hours - skipping
2019-10-22T11:31:56.001Z - error: [ui] Error checking for updates
Error: GitHub Desktop was not able to contact the update server. Ensure you have internet connectivity and try again.
    at Object.t.parseError (file:///C:/Users/sguarino/AppData/Local/GitHubDesktop/app-2.2.1/resources/app/renderer.js:20:123052)
    at onAutoUpdaterError (file:///C:/Users/sguarino/AppData/Local/GitHubDesktop/app-2.2.1/resources/app/renderer.js:12:99864)
    at CallbacksRegistry.apply (C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\callbacks-registry.js:47:25)
    at C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\api\remote.js:287:21
    at EventEmitter.<anonymous> (C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\api\remote.js:277:7)
    at EventEmitter.emit (events.js:194:13)
2019-10-22T11:31:56.975Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.649s)
2019-10-22T11:31:59.674Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T11:43:32.352Z - warn: [ui] getFetchPollInterval: failed for sarahgg/JavaProgrammingMasterclassByTimBuchalka
TypeError: Failed to fetch
2019-10-22T12:02:05.300Z - info: [ui] [BranchPruner] Last prune took place in 12 hours - skipping
2019-10-22T12:02:05.648Z - warn: [ui] fetchRepository: an error occurred for 'SarahGG/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T12:02:05.649Z - warn: [ui] fetchRepository: an error occurred for 'sarahgg/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T12:02:05.650Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: JavaProgrammingMasterclassByTimBuchalka - SarahGG (has token)
2019-10-22T12:02:05.879Z - error: [ui] `git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'https://github.com/SarahGG/JavaProgrammingMasterclassByTimBuchalka.git/': Could not resolve host: github.com

2019-10-22T12:02:05.954Z - info: [ui] [fetchProtectedBranches] unable to list protected branches
TypeError: Failed to fetch
2019-10-22T12:02:06.460Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T13:36:03.064Z - warn: [ui] fetchRepository: an error occurred for 'SarahGG/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T13:36:03.067Z - warn: [ui] fetchRepository: an error occurred for 'sarahgg/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T13:36:03.068Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: JavaProgrammingMasterclassByTimBuchalka - SarahGG (has token)
2019-10-22T13:36:03.150Z - error: [ui] `git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'https://github.com/SarahGG/JavaProgrammingMasterclassByTimBuchalka.git/': Could not resolve host: github.com

2019-10-22T13:36:03.221Z - error: [ui] Error checking for updates
Error: GitHub Desktop was not able to contact the update server. Ensure you have internet connectivity and try again.
    at Object.t.parseError (file:///C:/Users/sguarino/AppData/Local/GitHubDesktop/app-2.2.1/resources/app/renderer.js:20:123052)
    at onAutoUpdaterError (file:///C:/Users/sguarino/AppData/Local/GitHubDesktop/app-2.2.1/resources/app/renderer.js:12:99864)
    at CallbacksRegistry.apply (C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\callbacks-registry.js:47:25)
    at C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\api\remote.js:287:21
    at EventEmitter.<anonymous> (C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\resources\electron.asar\renderer\api\remote.js:277:7)
    at EventEmitter.emit (events.js:194:13)
2019-10-22T13:36:03.279Z - info: [ui] [fetchProtectedBranches] unable to list protected branches
TypeError: Failed to fetch
2019-10-22T13:36:03.829Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T13:36:03.829Z - warn: [ui] getFetchPollInterval: failed for sarahgg/JavaProgrammingMasterclassByTimBuchalka
TypeError: Failed to fetch
2019-10-22T13:39:31.353Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T13:44:39.014Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T14:36:27.091Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.091s)
2019-10-22T14:36:28.126Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T15:08:48.709Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T15:41:09.590Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T15:50:50.901Z - info: [ui] [BranchPruner] Last prune took place in 8 hours - skipping
2019-10-22T16:24:34.773Z - warn: [ui] fetchRepository: an error occurred for 'SarahGG/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T16:24:34.778Z - warn: [ui] fetchRepository: an error occurred for 'sarahgg/JavaProgrammingMasterclassByTimBuchalka'
TypeError: Failed to fetch
2019-10-22T16:24:34.778Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: JavaProgrammingMasterclassByTimBuchalka - SarahGG (has token)
2019-10-22T16:24:34.815Z - error: [ui] `git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'https://github.com/SarahGG/JavaProgrammingMasterclassByTimBuchalka.git/': Could not resolve host: github.com

2019-10-22T16:24:34.908Z - info: [ui] [fetchProtectedBranches] unable to list protected branches
TypeError: Failed to fetch
2019-10-22T16:24:35.599Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T16:24:35.600Z - warn: [ui] getFetchPollInterval: failed for sarahgg/JavaProgrammingMasterclassByTimBuchalka
TypeError: Failed to fetch
2019-10-22T17:17:38.800Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T17:25:01.229Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T17:57:21.883Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T18:29:43.042Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:02:02.953Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:34:22.354Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.071s)
2019-10-22T19:34:23.287Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:50:51.176Z - info: [ui] [BranchPruner] Last prune took place in 4 hours - skipping
2019-10-22T19:56:22.407Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:56:42.339Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:57:38.177Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:57:38.452Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T19:57:40.377Z - info: [ui] Executing isUsingLFS: git lfs track (took 1.813s)
2019-10-22T19:57:41.778Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:58:09.793Z - info: [ui] [Timing] Action 'create commit' for 'sguarino' took 9.616s
2019-10-22T19:58:09.833Z - info: [ui] [Timing] Action 'create commit' for 'sguarino' took 7.377s
2019-10-22T19:58:16.748Z - info: [ui] [Timing] Action 'create commit' for 'sguarino' took 7.838s
2019-10-22T19:58:16.758Z - info: [ui] [Timing] Action 'create commit' for 'sguarino' took 8.379s
2019-10-22T19:58:16.775Z - info: [ui] [Timing] Action 'create commit' for 'sguarino' took 7.711s
2019-10-22T19:58:37.824Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:58:52.295Z - info: [ui] [BranchPruner] Last prune took place in 4 hours - skipping
2019-10-22T19:58:52.936Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:00.306Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:00.794Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T19:59:02.996Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:27.916Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:31.990Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:31.990Z - info: [ui] Background fetch for 2 repositories took 2.587sec
2019-10-22T19:59:41.025Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T19:59:43.373Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:43.536Z - info: [ui] [BranchPruner] No branches to prune.
2019-10-22T19:59:43.656Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:47.879Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T19:59:51.352Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:00:24.061Z - info: [ui] [AppStore] loading 2 repositories from store
2019-10-22T20:00:24.062Z - info: [ui] [AppStore] found account: SarahGG (Sarah Guarino)
2019-10-22T20:00:24.284Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2019-10-22T20:00:24.972Z - info: [ui] launching: 2.2.1 (Windows 10.0.18362)
2019-10-22T20:00:24.972Z - info: [ui] execPath: 'C:\Users\sguarino\AppData\Local\GitHubDesktop\app-2.2.1\GitHubDesktop.exe'
2019-10-22T20:00:25.099Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:00:25.246Z - info: [ui] Stats reported.
2019-10-22T20:00:38.549Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:00:56.224Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:00:56.407Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.087s)
2019-10-22T20:00:56.847Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:07.657Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2019-10-22T20:01:08.328Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:20.696Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:39.417Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:43.483Z - info: [ui] [BranchPruner] No branches to prune.
2019-10-22T20:01:43.728Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:51.172Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:01:51.790Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T20:01:53.706Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:02:26.446Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:02:26.446Z - info: [ui] Background fetch for 2 repositories took 2.321sec
2019-10-22T20:04:32.006Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:04:41.656Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2019-10-22T20:04:42.225Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:04:43.046Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.315s)
2019-10-22T20:04:43.584Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:05:04.084Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:05:31.017Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:06:02.958Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:06:11.171Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:06:17.452Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino\Documents\GitHub\WebDeveloperBootcampByColtSteele\WebDeveloperBootcampByColtSteele to store
2019-10-22T20:06:19.125Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:06:28.052Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:07:22.231Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:07:35.102Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:07:42.771Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:07:43.053Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T20:07:45.037Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:08:32.231Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.316s)
2019-10-22T20:08:32.521Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:08:36.429Z - info: [ui] [BranchPruner] No branches to prune.
2019-10-22T20:08:36.626Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:08:40.328Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:09:22.109Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:11:07.002Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:13:23.037Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:13:56.768Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:14:00.194Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:14:12.257Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:17:28.152Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:17:28.152Z - info: [ui] Background fetch for 2 repositories took 1.695sec
2019-10-22T20:28:39.096Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:30:05.533Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:30:14.769Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:30:28.938Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:31:35.284Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store
2019-10-22T20:31:35.623Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:31:37.039Z - info: [ui] Executing isUsingLFS: git lfs track (took 1.645s)
2019-10-22T20:31:38.327Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:32:27.493Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.019s)
2019-10-22T20:32:28.116Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:32:28.447Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:32:30.467Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:32:30.575Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.298s)
2019-10-22T20:32:30.995Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:32:30.996Z - info: [ui] Background fetch for 3 repositories took 4.536sec
2019-10-22T20:32:42.830Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:33:41.074Z - info: [ui] Current tutorial step is NotApplicable
2019-10-22T20:39:38.579Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.090s)
2019-10-22T20:39:39.213Z - info: [ui] Current tutorial step is NotApplicable
```

**Comment by tierninho** [comment]
@sagaragarwal94 Thanks for the detailed issue. I tried reproducing this but no luck.

For whatever reason, the `2019-10-22T20:01:51.790Z - info: [ui] [AppStore] adding repository at C:\Users\sguarino to store` was added to Desktop somehow.

When creating a repo, try using the Create New Repo button (ctrl-N) and select your desired local path. Let us know how it goes.

**Comment by SarahGG** [comment]
When I try to create a new repo and select that folder, it won't allow me to complete it and instead tells me to use the 'add local repo' option that is giving me the issue.

I only started having this issue with the new desktop program - the old one did not have this issue.

**Comment by steveward** [comment]
@SarahGG is it possible there's a `.git` directory in your root user folder? Try running `ls -al ~` from the terminal and see if there is a `.git directory in the output.

**Comment by SarahGG** [comment]
<img width="713" alt="AnswerPicGitInRoot" src="https://user-images.githubusercontent.com/23173393/67433057-7017bb80-f5b5-11e9-917e-9b3ab6509835.png">

This is what I'm seeing. It appears there isn't a .git file in my root folder, but I have deleted one out of there previous to your request.

It appeared after the error occurred that I reported to you here. I deleted it as soon as I saw it. Trying to add my local repo to GitHub through the GUI presents the error and re-adds the .git file to my root folder (even when verifying that a different folder is selected). 

If I do nothing in the GitHub GUI, the error and the .git file in the root never occur. I can use Git through my IDEs, through the website, and through the shell just fine, and it causes neither issue. When I am home again, I'm going to try it on my desktop to see if this issue is replicated across two separate machines/wifi signals.

**Comment by tierninho** [comment]
> When I try to create a new repo and select that folder, it won't allow me to complete it and instead tells me to use the 'add local repo' option

Any chance you have a screenshot of this? Thanks.

**Comment by SarahGG** [comment]
This is a feature built into the GUI - it won't let me create a directory that already exists.

<img width="393" alt="CannotCreateLocalRepoAsNew" src="https://user-images.githubusercontent.com/23173393/67596180-25717d00-f736-11e9-90f9-bb86701fdf2f.png">

**Comment by tierninho** [comment]
There must be a git directory still present at that path, else you should be able to create a new repo there. You deleted the repo out of Desktop that had all you files? 

If you click "add this repository" you will be back in the same loop, ie adding all your files into Desktop. 

In File Explorer, click View > show Hidden files to see if there is remnants of a directory there.

**Comment by SarahGG** [comment]
This is on my work computer, so I do not have the admin access to view hidden files. 

I will say that I have deleted the git files out of my root directory many many times, and that it always adds them back. This happens when I try to add in a local repo, regardless of what or where that repo is. It will ALWAYS simply add the git files to my root instead of the folder that I have selected, then target those new git files in the root, and completely ignore the directory I tried to add.

Whether there were already git files there to begin with or not, it shouldn't be doing that. I have had 3 senior devs on my team who use this GUI verify that this is behaving abnormally.
